### PR TITLE
Meta: Switch to clang-format-14 as the standard formatter

### PR DESCRIPTION
**Pull request by IdanHo**

### CODE REVIEW

File: AK/Format.h [hunk 1]
```diff
@@ -407,14 +407,14 @@ struct Formatter<const char*> : Formatter<StringView> {
...
 };
 template<>
-struct Formatter<char*> : Formatter<const char*> {
+struct Formatter<char*> : Formatter<char const*> {
 };
 template<size_t Size>
-struct Formatter<char[Size]> : Formatter<const char*> {
-};
+struct Formatter<char[Size]> : Formatter<char const*> {};
+// clang-format off
```
Comment (BertalanD) [inline AK/Format.h:414]: This will be fixed in 14.0.0:
https://github.com/llvm/llvm-project/issues/53567

File: AK/StdLibExtraDetails.h [hunk 2]
```diff
@@ -562,11 +562,13 @@ struct __decay {
     typedef Detail::RemoveCVReference<T> type;
 };
 template<typename T>
-struct __decay<T[]> {
+struct __decay<T[]>
+{
```
Comment (BertalanD) [inline AK/StdLibExtraDetails.h:566]: This doesn't look quite right.

File: AK/Format.h (hunk 1 above)
Comment (IdanHo) [reply AK/Format.h:414]: Yeah, waiting until 14.0.0 is probably the best bet, I didn't know this was still an rc

**Comment by IdanHo** [comment]
Not stale, awaiting the full release of LLVM/Clang 14 on March 15.

**Comment by IdanHo** [comment]
Still not stale, waiting for Debian/Ubuntu clang-format-14 packages to be released

**Review by bgianfo — APPROVED** [review]
🎉🎉

**Pull request merged**